include/odp/api/spec/traffic_mngr.h

```diff
@@ -625,6 +625,34 @@ odp_tm_t odp_tm_find(const char            *name,
  */
 int odp_tm_capability(odp_tm_t odp_tm, odp_tm_capabilities_t *capabilities);
 
+/** Start a TM system.
+ *
+ * odp_tm_start() needs to be used to start an already created / found TM system
+ * and in stopped state. By default platforms can choose to be already in
+ * started state with immediately after created. Calling this function
+ * on an already started TM system will be ignored.
```

Comment:
Maybe it would be better to require all implementations to behave similarly, even if that causes slight backward compatibility. So how about requiring odp_tm_start() to be called and having the TM always in stopped state before that? In any case, the expected behaviour in stopped state probably should be described better.

That said, I wonder if this TM specific start & stop is needed or whether the pktio start & stop would be sufficient for this?

Reply:
Yes. But I was thinking whether a TM specific start/stop is needed or if pktio start/stop could be used for that. I.e. if a system does not support dynamic TM update, then the pktio needs to be put to the stopped state for the update. But I suppose there are a couple of things where a separate TM start/stop is useful: pktio stop will stop pktin too, whereas TM stop can only stop the TX. And not all TM egresses are necessarily pktouts.

Reply:
Agreed, PKTIO start/stop enforcement will work for TM with PKTIO egresses but will stop both pktin and pktout.
Our platform cannot however support TM non pktio egress. So you can take a call on whether we can have TM start and TM stop or enforce odp_pktio_start()/stop().

Reply:
As discussed, let's have the TM specific start & stop functions.